handle_linux.go

```diff
@@ -9,8 +9,11 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-// Empty handle used by the netlink package methods
-var pkgHandle = &Handle{}
+// Default handle used by the netlink package methods
+var pkgHandle *Handle = nil
```

Comment:
Why do we need to make this a pointer?
Can't we still set the socket TO in _init()_?

Reply:
Hi @aboch thanks for reviewing!

Not sure if I understand your question corrctly: if `pkgHandle` is initialized as `&Handle{}` as previously, the `pkgHandle.sockets` will be an empty map, and we can not set TO values for these sockets.

Reply:
Thanks @ArturChiao now I remember.

With your proposed change, any binary depending on _vishvananda/netlink_ will open 3 nl sockets, even when no netlink request is actually being executed (https://github.com/vishvananda/netlink/blob/v1.1.0/nl/nl_linux.go#L34).

Current behavior of this library has been: Open the nl socket only when you execute a pkg method (AddAddr(), AddLink(),...).

The addition of the _Handle_ thing was done in a way so that above behavior was not modified.

I know it is limiting, but have you considered setting a default TO in `func (req *NetlinkRequest) Execute()` at https://github.com/vishvananda/netlink/blob/v1.1.0/nl/nl_linux.go#L406 instead ?

At the end of the day, a minute timeout (the default you are proposing) should satisfy most if not all use-cases.

Reply:
Thanks for the explanations!  @aboch 

Set TO for [each socket](https://github.com/vishvananda/netlink/blob/v1.1.0/nl/nl_linux.go#L402) we're going to use makes sense to me.
But I'm not sure if you would like to **make the TO value configurable or not** (e.g. hardcode 1 minute).

Reply:
Refactored to set TO after we got a socket, please have a review, thanks! @aboch

Reply:
PTAL @aboch